{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 18\u001b[0m\n\u001b[0;32m     16\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39m__name__\u001b[39m \u001b[39m==\u001b[39m \u001b[39m'\u001b[39m\u001b[39m__main__\u001b[39m\u001b[39m'\u001b[39m:\n\u001b[0;32m     17\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m---> 18\u001b[0m         local2cloud()\n\u001b[0;32m     19\u001b[0m     \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m:\n\u001b[0;32m     20\u001b[0m         \u001b[39mpass\u001b[39;00m\n",
      "Cell \u001b[1;32mIn[5], line 14\u001b[0m, in \u001b[0;36mlocal2cloud\u001b[1;34m()\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[39mwhile\u001b[39;00m(\u001b[39mTrue\u001b[39;00m):\n\u001b[0;32m     13\u001b[0m     \u001b[39mfor\u001b[39;00m collection \u001b[39min\u001b[39;00m col\u001b[39m.\u001b[39mCollections:\n\u001b[1;32m---> 14\u001b[0m             c2l\u001b[39m.\u001b[39;49mdownloadBase(database\u001b[39m=\u001b[39;49mdb\u001b[39m.\u001b[39;49mdbBuffer, collection\u001b[39m=\u001b[39;49mcollection[\u001b[39m'\u001b[39;49m\u001b[39mname\u001b[39;49m\u001b[39m'\u001b[39;49m])\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\DatabaseSync\\GlobalSets\\c2l_functions.py:48\u001b[0m, in \u001b[0;36mdownloadBase\u001b[1;34m(database, collection)\u001b[0m\n\u001b[0;32m     37\u001b[0m         documents \u001b[39m=\u001b[39m cloud\u001b[39m.\u001b[39maggregate([\n\u001b[0;32m     38\u001b[0m             {\n\u001b[0;32m     39\u001b[0m                 \u001b[39m'\u001b[39m\u001b[39m$sort\u001b[39m\u001b[39m'\u001b[39m: {\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     45\u001b[0m             }\n\u001b[0;32m     46\u001b[0m         ])\n\u001b[0;32m     47\u001b[0m         \u001b[39mwhile\u001b[39;00m documents\u001b[39m.\u001b[39m_has_next():\n\u001b[1;32m---> 48\u001b[0m             downloadElement(document\u001b[39m=\u001b[39;49mdocuments\u001b[39m.\u001b[39;49mnext(), local\u001b[39m=\u001b[39;49mlocal, cloud\u001b[39m=\u001b[39;49mcloud)\n\u001b[0;32m     50\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m \u001b[39mas\u001b[39;00m e:\n\u001b[0;32m     51\u001b[0m     eStr \u001b[39m=\u001b[39m \u001b[39mstr\u001b[39m(e)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\DatabaseSync\\GlobalSets\\c2l_functions.py:14\u001b[0m, in \u001b[0;36mdownloadElement\u001b[1;34m(document, local, cloud)\u001b[0m\n\u001b[0;32m     12\u001b[0m     \u001b[39mfor\u001b[39;00m upTry \u001b[39min\u001b[39;00m \u001b[39mrange\u001b[39m(\u001b[39m0\u001b[39m,\u001b[39m5\u001b[39m):\n\u001b[0;32m     13\u001b[0m         \u001b[39mif\u001b[39;00m MongoClient\u001b[39m.\u001b[39mLocalClient\u001b[39m.\u001b[39mis_primary \u001b[39mand\u001b[39;00m local\u001b[39m.\u001b[39minsert_one(document)\u001b[39m.\u001b[39macknowledged:\n\u001b[1;32m---> 14\u001b[0m             cloud\u001b[39m.\u001b[39;49mdelete_one(document)\n\u001b[0;32m     15\u001b[0m             \u001b[39mbreak\u001b[39;00m\n\u001b[0;32m     16\u001b[0m \u001b[39mexcept\u001b[39;00m errors\u001b[39m.\u001b[39mDuplicateKeyError:\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\collection.py:1248\u001b[0m, in \u001b[0;36mCollection.delete_one\u001b[1;34m(self, filter, collation, hint, session)\u001b[0m\n\u001b[0;32m   1211\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Delete a single document matching the filter.\u001b[39;00m\n\u001b[0;32m   1212\u001b[0m \n\u001b[0;32m   1213\u001b[0m \u001b[39m  >>> db.test.count_documents({'x': 1})\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1244\u001b[0m \u001b[39m.. versionadded:: 3.0\u001b[39;00m\n\u001b[0;32m   1245\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m   1246\u001b[0m write_concern \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_write_concern_for(session)\n\u001b[0;32m   1247\u001b[0m \u001b[39mreturn\u001b[39;00m DeleteResult(\n\u001b[1;32m-> 1248\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_delete_retryable(\n\u001b[0;32m   1249\u001b[0m         \u001b[39mfilter\u001b[39;49m, \u001b[39mFalse\u001b[39;49;00m,\n\u001b[0;32m   1250\u001b[0m         write_concern\u001b[39m=\u001b[39;49mwrite_concern,\n\u001b[0;32m   1251\u001b[0m         collation\u001b[39m=\u001b[39;49mcollation, hint\u001b[39m=\u001b[39;49mhint, session\u001b[39m=\u001b[39;49msession),\n\u001b[0;32m   1252\u001b[0m     write_concern\u001b[39m.\u001b[39macknowledged)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\collection.py:1206\u001b[0m, in \u001b[0;36mCollection._delete_retryable\u001b[1;34m(self, criteria, multi, write_concern, op_id, ordered, collation, hint, session)\u001b[0m\n\u001b[0;32m   1199\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_delete\u001b[39m(session, sock_info, retryable_write):\n\u001b[0;32m   1200\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_delete(\n\u001b[0;32m   1201\u001b[0m         sock_info, criteria, multi,\n\u001b[0;32m   1202\u001b[0m         write_concern\u001b[39m=\u001b[39mwrite_concern, op_id\u001b[39m=\u001b[39mop_id, ordered\u001b[39m=\u001b[39mordered,\n\u001b[0;32m   1203\u001b[0m         collation\u001b[39m=\u001b[39mcollation, hint\u001b[39m=\u001b[39mhint, session\u001b[39m=\u001b[39msession,\n\u001b[0;32m   1204\u001b[0m         retryable_write\u001b[39m=\u001b[39mretryable_write)\n\u001b[1;32m-> 1206\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__database\u001b[39m.\u001b[39;49mclient\u001b[39m.\u001b[39;49m_retryable_write(\n\u001b[0;32m   1207\u001b[0m     (write_concern \u001b[39mor\u001b[39;49;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mwrite_concern)\u001b[39m.\u001b[39;49macknowledged \u001b[39mand\u001b[39;49;00m \u001b[39mnot\u001b[39;49;00m multi,\n\u001b[0;32m   1208\u001b[0m     _delete, session)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\mongo_client.py:1552\u001b[0m, in \u001b[0;36mMongoClient._retryable_write\u001b[1;34m(self, retryable, func, session)\u001b[0m\n\u001b[0;32m   1550\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Internal retryable write helper.\"\"\"\u001b[39;00m\n\u001b[0;32m   1551\u001b[0m \u001b[39mwith\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_tmp_session(session) \u001b[39mas\u001b[39;00m s:\n\u001b[1;32m-> 1552\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_retry_with_session(retryable, func, s, \u001b[39mNone\u001b[39;49;00m)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\mongo_client.py:1438\u001b[0m, in \u001b[0;36mMongoClient._retry_with_session\u001b[1;34m(self, retryable, func, session, bulk)\u001b[0m\n\u001b[0;32m   1429\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Execute an operation with at most one consecutive retries\u001b[39;00m\n\u001b[0;32m   1430\u001b[0m \n\u001b[0;32m   1431\u001b[0m \u001b[39mReturns func()'s return value on success. On error retries the same\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1434\u001b[0m \u001b[39mRe-raises any exception thrown by func().\u001b[39;00m\n\u001b[0;32m   1435\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m   1436\u001b[0m retryable \u001b[39m=\u001b[39m (retryable \u001b[39mand\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mretry_writes\n\u001b[0;32m   1437\u001b[0m              \u001b[39mand\u001b[39;00m session \u001b[39mand\u001b[39;00m \u001b[39mnot\u001b[39;00m session\u001b[39m.\u001b[39min_transaction)\n\u001b[1;32m-> 1438\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_retry_internal(retryable, func, session, bulk)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\mongo_client.py:1470\u001b[0m, in \u001b[0;36mMongoClient._retry_internal\u001b[1;34m(self, retryable, func, session, bulk)\u001b[0m\n\u001b[0;32m   1468\u001b[0m                 \u001b[39mraise\u001b[39;00m last_error\n\u001b[0;32m   1469\u001b[0m             retryable \u001b[39m=\u001b[39m \u001b[39mFalse\u001b[39;00m\n\u001b[1;32m-> 1470\u001b[0m         \u001b[39mreturn\u001b[39;00m func(session, sock_info, retryable)\n\u001b[0;32m   1471\u001b[0m \u001b[39mexcept\u001b[39;00m ServerSelectionTimeoutError:\n\u001b[0;32m   1472\u001b[0m     \u001b[39mif\u001b[39;00m is_retrying():\n\u001b[0;32m   1473\u001b[0m         \u001b[39m# The application may think the write was never attempted\u001b[39;00m\n\u001b[0;32m   1474\u001b[0m         \u001b[39m# if we raise ServerSelectionTimeoutError on the retry\u001b[39;00m\n\u001b[0;32m   1475\u001b[0m         \u001b[39m# attempt. Raise the original exception instead.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\collection.py:1200\u001b[0m, in \u001b[0;36mCollection._delete_retryable.<locals>._delete\u001b[1;34m(session, sock_info, retryable_write)\u001b[0m\n\u001b[0;32m   1199\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_delete\u001b[39m(session, sock_info, retryable_write):\n\u001b[1;32m-> 1200\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_delete(\n\u001b[0;32m   1201\u001b[0m         sock_info, criteria, multi,\n\u001b[0;32m   1202\u001b[0m         write_concern\u001b[39m=\u001b[39;49mwrite_concern, op_id\u001b[39m=\u001b[39;49mop_id, ordered\u001b[39m=\u001b[39;49mordered,\n\u001b[0;32m   1203\u001b[0m         collation\u001b[39m=\u001b[39;49mcollation, hint\u001b[39m=\u001b[39;49mhint, session\u001b[39m=\u001b[39;49msession,\n\u001b[0;32m   1204\u001b[0m         retryable_write\u001b[39m=\u001b[39;49mretryable_write)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\collection.py:1183\u001b[0m, in \u001b[0;36mCollection._delete\u001b[1;34m(self, sock_info, criteria, multi, write_concern, op_id, ordered, collation, hint, session, retryable_write)\u001b[0m\n\u001b[0;32m   1176\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_legacy_write(\n\u001b[0;32m   1177\u001b[0m         sock_info, \u001b[39m'\u001b[39m\u001b[39mdelete\u001b[39m\u001b[39m'\u001b[39m, command, op_id,\n\u001b[0;32m   1178\u001b[0m         \u001b[39mFalse\u001b[39;00m, message\u001b[39m.\u001b[39mdelete, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m__full_name, criteria,\n\u001b[0;32m   1179\u001b[0m         \u001b[39mFalse\u001b[39;00m, write_concern\u001b[39m.\u001b[39mdocument,\n\u001b[0;32m   1180\u001b[0m         \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m__write_response_codec_options,\n\u001b[0;32m   1181\u001b[0m         \u001b[39mint\u001b[39m(\u001b[39mnot\u001b[39;00m multi))\n\u001b[0;32m   1182\u001b[0m \u001b[39m# Delete command.\u001b[39;00m\n\u001b[1;32m-> 1183\u001b[0m result \u001b[39m=\u001b[39m sock_info\u001b[39m.\u001b[39;49mcommand(\n\u001b[0;32m   1184\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__database\u001b[39m.\u001b[39;49mname,\n\u001b[0;32m   1185\u001b[0m     command,\n\u001b[0;32m   1186\u001b[0m     write_concern\u001b[39m=\u001b[39;49mwrite_concern,\n\u001b[0;32m   1187\u001b[0m     codec_options\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__write_response_codec_options,\n\u001b[0;32m   1188\u001b[0m     session\u001b[39m=\u001b[39;49msession,\n\u001b[0;32m   1189\u001b[0m     client\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__database\u001b[39m.\u001b[39;49mclient,\n\u001b[0;32m   1190\u001b[0m     retryable_write\u001b[39m=\u001b[39;49mretryable_write)\n\u001b[0;32m   1191\u001b[0m _check_write_command_response(result)\n\u001b[0;32m   1192\u001b[0m \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\pool.py:726\u001b[0m, in \u001b[0;36mSocketInfo.command\u001b[1;34m(self, dbname, spec, secondary_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events, user_fields, exhaust_allowed)\u001b[0m\n\u001b[0;32m    724\u001b[0m \u001b[39m# Catch socket.error, KeyboardInterrupt, etc. and close ourselves.\u001b[39;00m\n\u001b[0;32m    725\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mBaseException\u001b[39;00m \u001b[39mas\u001b[39;00m error:\n\u001b[1;32m--> 726\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_raise_connection_failure(error)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\pool.py:710\u001b[0m, in \u001b[0;36mSocketInfo.command\u001b[1;34m(self, dbname, spec, secondary_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events, user_fields, exhaust_allowed)\u001b[0m\n\u001b[0;32m    708\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_raise_if_not_writable(unacknowledged)\n\u001b[0;32m    709\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m--> 710\u001b[0m     \u001b[39mreturn\u001b[39;00m command(\u001b[39mself\u001b[39;49m, dbname, spec, secondary_ok,\n\u001b[0;32m    711\u001b[0m                    \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mis_mongos, read_preference, codec_options,\n\u001b[0;32m    712\u001b[0m                    session, client, check, allowable_errors,\n\u001b[0;32m    713\u001b[0m                    \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49maddress, check_keys, listeners,\n\u001b[0;32m    714\u001b[0m                    \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mmax_bson_size, read_concern,\n\u001b[0;32m    715\u001b[0m                    parse_write_concern_error\u001b[39m=\u001b[39;49mparse_write_concern_error,\n\u001b[0;32m    716\u001b[0m                    collation\u001b[39m=\u001b[39;49mcollation,\n\u001b[0;32m    717\u001b[0m                    compression_ctx\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mcompression_context,\n\u001b[0;32m    718\u001b[0m                    use_op_msg\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mop_msg_enabled,\n\u001b[0;32m    719\u001b[0m                    unacknowledged\u001b[39m=\u001b[39;49munacknowledged,\n\u001b[0;32m    720\u001b[0m                    user_fields\u001b[39m=\u001b[39;49muser_fields,\n\u001b[0;32m    721\u001b[0m                    exhaust_allowed\u001b[39m=\u001b[39;49mexhaust_allowed)\n\u001b[0;32m    722\u001b[0m \u001b[39mexcept\u001b[39;00m (OperationFailure, NotPrimaryError):\n\u001b[0;32m    723\u001b[0m     \u001b[39mraise\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\network.py:152\u001b[0m, in \u001b[0;36mcommand\u001b[1;34m(sock_info, dbname, spec, secondary_ok, is_mongos, read_preference, codec_options, session, client, check, allowable_errors, address, check_keys, listeners, max_bson_size, read_concern, parse_write_concern_error, collation, compression_ctx, use_op_msg, unacknowledged, user_fields, exhaust_allowed)\u001b[0m\n\u001b[0;32m    150\u001b[0m     response_doc \u001b[39m=\u001b[39m {\u001b[39m\"\u001b[39m\u001b[39mok\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39m1\u001b[39m}\n\u001b[0;32m    151\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m--> 152\u001b[0m     reply \u001b[39m=\u001b[39m receive_message(sock_info, request_id)\n\u001b[0;32m    153\u001b[0m     sock_info\u001b[39m.\u001b[39mmore_to_come \u001b[39m=\u001b[39m reply\u001b[39m.\u001b[39mmore_to_come\n\u001b[0;32m    154\u001b[0m     unpacked_docs \u001b[39m=\u001b[39m reply\u001b[39m.\u001b[39munpack_response(\n\u001b[0;32m    155\u001b[0m         codec_options\u001b[39m=\u001b[39mcodec_options, user_fields\u001b[39m=\u001b[39muser_fields)\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\network.py:200\u001b[0m, in \u001b[0;36mreceive_message\u001b[1;34m(sock_info, request_id, max_message_size)\u001b[0m\n\u001b[0;32m    197\u001b[0m     deadline \u001b[39m=\u001b[39m \u001b[39mNone\u001b[39;00m\n\u001b[0;32m    198\u001b[0m \u001b[39m# Ignore the response's request id.\u001b[39;00m\n\u001b[0;32m    199\u001b[0m length, _, response_to, op_code \u001b[39m=\u001b[39m _UNPACK_HEADER(\n\u001b[1;32m--> 200\u001b[0m     _receive_data_on_socket(sock_info, \u001b[39m16\u001b[39;49m, deadline))\n\u001b[0;32m    201\u001b[0m \u001b[39m# No request_id for exhaust cursor \"getMore\".\u001b[39;00m\n\u001b[0;32m    202\u001b[0m \u001b[39mif\u001b[39;00m request_id \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\andre\\GitHub\\Thesis\\backend\\django\\venv\\lib\\site-packages\\pymongo\\network.py:291\u001b[0m, in \u001b[0;36m_receive_data_on_socket\u001b[1;34m(sock_info, length, deadline)\u001b[0m\n\u001b[0;32m    289\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m    290\u001b[0m     wait_for_read(sock_info, deadline)\n\u001b[1;32m--> 291\u001b[0m     chunk_length \u001b[39m=\u001b[39m sock_info\u001b[39m.\u001b[39;49msock\u001b[39m.\u001b[39;49mrecv_into(mv[bytes_read:])\n\u001b[0;32m    292\u001b[0m \u001b[39mexcept\u001b[39;00m (\u001b[39mIOError\u001b[39;00m, \u001b[39mOSError\u001b[39;00m) \u001b[39mas\u001b[39;00m exc:\n\u001b[0;32m    293\u001b[0m     \u001b[39mif\u001b[39;00m _errno_from_exception(exc) \u001b[39m==\u001b[39m errno\u001b[39m.\u001b[39mEINTR:\n",
      "File \u001b[1;32mC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\\lib\\ssl.py:1242\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[1;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[0;32m   1238\u001b[0m     \u001b[39mif\u001b[39;00m flags \u001b[39m!=\u001b[39m \u001b[39m0\u001b[39m:\n\u001b[0;32m   1239\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[0;32m   1240\u001b[0m           \u001b[39m\"\u001b[39m\u001b[39mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[0;32m   1241\u001b[0m           \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__class__\u001b[39m)\n\u001b[1;32m-> 1242\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mread(nbytes, buffer)\n\u001b[0;32m   1243\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m   1244\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39msuper\u001b[39m()\u001b[39m.\u001b[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001b[1;32mC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\\lib\\ssl.py:1100\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[1;34m(self, len, buffer)\u001b[0m\n\u001b[0;32m   1098\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m   1099\u001b[0m     \u001b[39mif\u001b[39;00m buffer \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m-> 1100\u001b[0m         \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sslobj\u001b[39m.\u001b[39;49mread(\u001b[39mlen\u001b[39;49m, buffer)\n\u001b[0;32m   1101\u001b[0m     \u001b[39melse\u001b[39;00m:\n\u001b[0;32m   1102\u001b[0m         \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_sslobj\u001b[39m.\u001b[39mread(\u001b[39mlen\u001b[39m)\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "# Global imports\n",
    "from GlobalSets.Mongo import Clients as MongoClient, DataBases as db, Collections as col\n",
    "import GlobalSets.c2l_functions as c2l\n",
    "\n",
    "# Import librarys\n",
    "from pymongo import collection, errors\n",
    "import bson\n",
    "\n",
    "def local2cloud():\n",
    "    while(True):\n",
    "        for collection in col.Collections:\n",
    "                c2l.downloadBase(database=db.dbBuffer, collection=collection['name'])\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    try:\n",
    "        local2cloud()\n",
    "    except Exception:\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "347bf08e745969db531fa1bdcede93f4381a1ecf9ee78f652807127caca4d300"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
